cookbooks/aws-parallelcluster-config/resources/mount_fsx.rb

```diff
@@ -105,3 +105,59 @@
     end
   end
 end
+
+action :unmount do
+  fsx_fs_id_array = new_resource.fsx_fs_id_array.dup
+  fsx_fs_type_array = new_resource.fsx_fs_type_array.dup
+  fsx_shared_dir_array = new_resource.fsx_shared_dir_array.dup
+  fsx_dns_name_array = new_resource.fsx_dns_name_array.dup
+  fsx_mount_name_array = new_resource.fsx_mount_name_array.dup
+  fsx_volume_junction_path_array = new_resource.fsx_volume_junction_path_array.dup
+
+  fsx_fs_id_array.each_with_index do |fsx_fs_id, index|
+    fsx_shared_dir = fsx_shared_dir_array[index]
+    fsx_dns_name = fsx_dns_name_array[index]
+    fsx_fs_type = fsx_fs_type_array[index]
+    fsx_volume_junction_path = fsx_volume_junction_path_array[index]
+
+    dns_name = if fsx_dns_name && !fsx_dns_name.empty?
+                 # DNS names of existing Lustre, Ontap, OpenZFS file systems are passed in from CLI
+                 fsx_dns_name
+               else
+                 # DNS names of newly created Lustre file systems are hardcoded here.
+                 # Note the Hardcoding format is only valid for lustre file systems created after Mar-1 2021
+                 "#{fsx_fs_id}.fsx.#{node['cluster']['region']}.amazonaws.com"
+               end
+    fsx_shared_dir = "/#{fsx_shared_dir}" unless fsx_shared_dir.start_with?('/')
+    execute 'unmount fsx' do
+      command "umount #{fsx_shared_dir}"
+      retries 3
+      retry_delay 5
```

Comment:
In other mounting/unmounting operations we used retries=10, retry_delay=6. Unless we have a specific reasoning for the current values, what about aligning these kind of operations to the same values?

Reply:
good idea, changed